drivers/interrupt_controller/Kconfig.plic

```diff
@@ -13,6 +13,44 @@ config PLIC
 
 if PLIC
 
+config PLIC_SUPPORTS_SOFT_INTERRUPT
+	bool
+	default y
+	depends on DT_HAS_ANDESTECH_NCEPLIC100_ENABLED
+	help
+	  Enabled when the PLIC supports software-triggered interrupts.
+
+config PLIC_SUPPORTS_TRIG_TYPE
+	bool
+	default y
+	depends on DT_HAS_ANDESTECH_NCEPLIC100_ENABLED
+	help
+	  Enabled when the PLIC supports multiple trigger types,
+	  such as level, edge, etc.
+
+if PLIC_SUPPORTS_TRIG_TYPE
+
+config PLIC_TRIG_TYPE_REG_OFFSET
+	hex "Trigger type register offset"
+	default 0x1080 if DT_HAS_ANDESTECH_NCEPLIC100_ENABLED
+	help
+	  Offset to the 'trigger type' register.
+
+config PLIC_TRIG_TYPE_WIDTH
+	int "Trigger type width"
+	default 1 if DT_HAS_ANDESTECH_NCEPLIC100_ENABLED
+	help
```

Comment:
I purposely left this blank so that the driver would fail to compile if some OOT configuration selected `PLIC_SUPPORTS_TRIG_TYPE` but forgot to configure these (same as `PLIC_TRIG_TYPE_REG_OFFSET`).

Is a default value necessary? If trigger type is supported, this should be >= 1

Reply:
Maybe make the default 1 then, independent of which plic implementation it is.

If other plic implementations have larger bit widths, then there should be default specialization